cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1374,22 +1387,99 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
          * There is one special case of scaling from standalone (single one) Zookeeper pod.
          * In this case quorum cannot be preserved.
          */
-        Future<ReconciliationState> zkScaleUpInSmallSteps() {
+        Future<ReconciliationState> zkScalingSetup() {
             return zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name))
-                    .map(sts -> sts == null ? Integer.valueOf(0) : sts.getSpec().getReplicas())
-                    .compose(currentReplicas -> {
-                        if (currentReplicas > 0 && zkCluster.getReplicas() > currentReplicas) {
-                            zkCluster.setReplicas(currentReplicas + 1);
-                            zkScalingUp = true;
-                        }
+                    .compose(sts -> {
+                        if (sts != null) {
+                            int currentReplicas = sts.getSpec().getReplicas();
+                            int desiredReplicas = zkCluster.getReplicas();
+                            int replicaDiff = Math.abs(currentReplicas - desiredReplicas);
+                            boolean isZK35x = KafkaVersion.compareDottedVersions(zkCluster.getVersion(), "3.4.99") > 0;
+
+                            if (isZK35x && replicaDiff > 1) {
+                                // If the user is scaling a 3.5.x cluster and has specified as scale up or down of more than one
+                                // then log the error and throw an exception halting the reconciliation.
+
+                                String scaleErrorMessage = String.format(
+                                        "Scaling of Zookeeper 3.5.x clusters must be done in steps of one only. " +
+                                                "A change of %d was specified. Please set the Zookeeper replica count " +
+                                                "to one greater or lesser than the original value (%d).",
+                                        replicaDiff, currentReplicas);
+
+                                log.error(scaleErrorMessage);
+
+                                return Future.failedFuture(new IllegalArgumentException(scaleErrorMessage));
+
+                            } else if (desiredReplicas > currentReplicas) {
+                                // If this is a scale up of a 3.4.x cluster we want to scale in steps of one at a time
```

Comment:
I'm not sure I understand the comment, because at this point you do not know this is 3.4 or 3.5 - it is not part of this particular `else if`, or?